File: java/messages/src/main/java/com/nedap/retail/messages/article/Article.java
package com.nedap.retail.messages.article;

import java.io.Serializable;
import java.util.List;
import java.util.UUID;

import org.apache.commons.lang.StringUtils;
import org.codehaus.jackson.map.annotate.JsonSerialize;
import org.joda.time.DateTime;

import com.fasterxml.jackson.annotation.JsonIgnore;

@JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
public class Article implements Serializable {

    private static final long serialVersionUID = 3681466505727698821L;

    public static final String GTIN = "gtin";
    public static final String STYLE = "style";
    public static final String OPTION = "option";
    public static final String COLOR = "color";
    public static final String SIZES = "sizes";
    public static final String PRICES = "prices";
    public static final String SUPPLIER = "supplier";
    public static final String CATEGORY = "category";
    public static final String CODE = "code";
    public static final String BRAND = "brand";
    public static final String SEASON = "season";
    public static final String NAME = "name";
    public static final String BARCODES = "barcodes";
    public static final String LAST_UPDATED = "last_updated";
    public static final String MARKDOWN = "markdown";

    public String gtin;
    public List<Barcode> barcodes;
    public String code;
    public String brand;
    public String season;
    public String name;
    public String option;
    public String style;
    public String color;
    public List<Size> sizes;
    public String supplier;
    public String category;
    public List<Price> prices;
    public Boolean markdown;

    @org.codehaus.jackson.annotate.JsonProperty(LAST_UPDATED)
    public DateTime lastUpdated;

    @JsonIgnore
    @org.codehaus.jackson.annotate.JsonIgnore
    public UUID lastUpdatedUUID;

    // Empty constructor used by Jackson
    public Article() {
    }

    public Article(final String gtin, final List<Barcode> barcodes, final String code, final String brand,
            final String season, final String name, final String option, final String style, final String color,
            final List<Size> sizes, final String supplier, final String category, final List<Price> prices,
            final DateTime lastUpdated) {
        this.gtin = gtin;
        this.barcodes = barcodes;
        this.code = code;
        this.brand = brand;
        this.season = season;
        this.name = name;
        this.option = option;
        this.style = style;
        this.color = color;
        this.sizes = sizes;
        this.supplier = supplier;
        this.category = category;
        this.prices = prices;
        this.lastUpdated = lastUpdated;
    }

    @Override
    public String toString() {
        return "Article [gtin=" + gtin + ", barcodes=" + StringUtils.join(barcodes, ";") + ", code=" + code
                + ", brand=" + brand + ", season=" + season + ", name=" + name + ", option=" + option + ", style="
                + style + ", color=" + color + ", sizes=" + StringUtils.join(sizes, ";") + ", supplier=" + supplier
                + ", category=" + category + ", prices=" + StringUtils.join(prices, ";") + ", last updated="
                + lastUpdated + "]";
    }

    @Override
    public int hashCode() {
        final int prime = 31;
        int result = 1;
        result = prime * result + (gtin == null ? 0 : gtin.hashCode());
        return result;
    }

    @Override
    public boolean equals(final Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        final Article other = (Article) obj;
        if (gtin == null) {
            if (other.gtin != null) {
                return false;
            }
        } else if (!gtin.equals(other.gtin)) {
            return false;
        }
        return true;
    }
}

File: csharp/NedapRetailApiClient/Nedap.Retail.Api/Article.V2/Article.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;

namespace Nedap.Retail.Api.Article.V2
{
    /// <summary>
    /// A description of an article that is visibly the same in terms of name, size, color, texture.
    /// GTIN is the leading identifier for each article description.
    /// </summary>
    public class Article : JsonPrintableObject
    {
        /// <summary>
        /// Optional. GTIN-14, length of 14 digits. Primary key to identify an article.
        /// GTIN-14 value is validated against length and can contain only digits.
        /// When no GTIN-14 is provided, it will be generated. It will be unique for this organization.
        /// </summary>
        [JsonProperty("gtin")]
        public string Gtin { get; set; }

        /// <summary>
        /// Contains all barcodes for this article, typically including EAN-13.
        /// Minimum of 1 linear barcode is required.
        /// </summary>
        [JsonProperty("barcodes")]
        public List<Barcode> Barcodes { get; set; }

        /// <summary>
        /// Optional. Article code as assigned by the retailer. No relation to GTIN or other barcodes is assumed.
        /// </summary>
        [JsonProperty("code")]
        public string Code { get; set; }

        /// <summary>
        /// Optional. The brand of the product, e.g. Nike or Adidas.
        /// </summary>
        [JsonProperty("brand")]
        public string Brand { get; set; }

        /// <summary>
        /// Optional. The season the product belongs to.
        /// Examples: 'Summer 2013' or 'Autumn 2014'.
        /// </summary>
        [JsonProperty("season")]
        public string Season { get; set; }

        /// <summary>
        /// Name of the article.
        /// </summary>
        [JsonProperty("name")]
        public string Name { get; set; }

        /// <summary>
        /// Identifies article in color.
        /// When omitted: option is automatically filled with name and color fields.
        /// </summary>
        [JsonProperty("option")]
        public string Option { get; set; }

        /// <summary>
        /// Identifies article, regardless of color.
        /// When omitted: style is automatically filled with name.
        /// </summary>
        [JsonProperty("style")]
        public string Style { get; set; }

        /// <summary>
        /// Contains color description.
        /// </summary>
        [JsonProperty("color")]
        public string Color { get; set; }

        /// <summary>
        /// Contains region dependent size descriptions.
        /// </summary>
        [JsonProperty("sizes")]
        public List<Size> Sizes { get; set; }

        /// <summary>
        /// Optional. Supplier name. Can also be factory name.
        /// </summary>
        [JsonProperty("supplier")]
        public string Supplier { get; set; }

        /// <summary>
        /// Optional. Can contain various prices that are country or region specific.
        /// </summary>
        [JsonProperty("prices")]
        public List<Price> Prices { get; set; }

        /// <summary>
        /// The date + time moment when article was last updated.
        /// </summary>
        [JsonProperty("last_updated")]
        public DateTime LastUpdated { get; set; }
    }
}